tests/aws/services/kms/test_kms.py

```diff
@@ -1465,6 +1465,159 @@ def test_rotate_key_on_demand_raises_error_given_key_with_imported_key_material(
+        snapshot.match("after-on-demand-rotation", response_after_on_demand_rotation)
+
+        assert (
+            len(response_after_on_demand_rotation["Rotations"])
+            == len(response_initial_rotation["Rotations"]) + 1
+        )
+
+    @markers.aws.validated
+    def test_list_key_rotations_different_key_states(self, kms_create_key, aws_client, snapshot):
+        key_id = kms_create_key(KeyUsage="ENCRYPT_DECRYPT", KeySpec="SYMMETRIC_DEFAULT")["KeyId"]
+
+        snapshot.add_transformer(
+            snapshot.transform.key_value("KeyMaterialId", reference_replacement=False)
+        )
+
+        response_enabled_key = aws_client.kms.list_key_rotations(
+            KeyId=key_id, IncludeKeyMaterial="ALL_KEY_MATERIAL"
+        )
+        snapshot.match("enabled-key-rotations", response_enabled_key)
+
+        aws_client.kms.disable_key(KeyId=key_id)
+        response_disabled_key = aws_client.kms.list_key_rotations(
+            KeyId=key_id, IncludeKeyMaterial="ALL_KEY_MATERIAL"
+        )
+        snapshot.match("disabled-key-rotations", response_disabled_key)
+
+        aws_client.kms.schedule_key_deletion(KeyId=key_id, PendingWindowInDays=7)
+        response_pending_deletion_key = aws_client.kms.list_key_rotations(
+            KeyId=key_id, IncludeKeyMaterial="ALL_KEY_MATERIAL"
+        )
+        snapshot.match("pending-deletion-key-rotations", response_pending_deletion_key)
+
+    @markers.aws.validated
+    def test_list_key_rotations_content(self, kms_create_key, aws_client, snapshot):
+        key_id = kms_create_key(KeyUsage="ENCRYPT_DECRYPT", KeySpec="SYMMETRIC_DEFAULT")["KeyId"]
+
+        snapshot.add_transformer(
+            snapshot.transform.key_value("KeyMaterialId", reference_replacement=False)
+        )
+
+        for _ in range(3):
+            aws_client.kms.rotate_key_on_demand(KeyId=key_id)
+
+            def _rotation_completed():
+                response_status = aws_client.kms.get_key_rotation_status(KeyId=key_id)
+                return "OnDemandRotationStartDate" not in response_status
+
+            assert poll_condition(condition=_rotation_completed, timeout=10, interval=1)
+
+        response_after_rotations = aws_client.kms.list_key_rotations(
+            KeyId=key_id, IncludeKeyMaterial="ALL_KEY_MATERIAL"
+        )
+        snapshot.match("multiple-rotations", response_after_rotations)
+
+        rotations = response_after_rotations["Rotations"]
+        assert len(rotations) >= 4  # 1 initial + 3 on-demand
+
+        for rotation in rotations:
+            assert "KeyId" in rotation
+            assert (
+                "RotationDate" in rotation or rotation.get("RotationDate") is None
+            )  # None in initial
+            assert (
+                "RotationType" in rotation or rotation.get("RotationType") is None
+            )  # None in initial
+            assert "KeyMaterialId" in rotation
+            assert "KeyMaterialState" in rotation
```

Comment:
The test suggests that it would validate the `list_key_rotations` content and it is doing so in snapshot `multiple-rotations`, why do we explicitly need the assert here then since we capture the exact same thing already or is there something i am missing here?